### PR TITLE
Simplify code without semantic changes

### PR DESCRIPTION
**Pull request by arcz**
This PR simplifies some of the most syntactically complex places in our code.

### CODE REVIEW

File: lib/Echidna/Exec.hs [hunk 1]
```diff
@@ -66,23 +66,26 @@ vmExcept e = throwM $ case VMFailure e of {Illegal -> IllegalExec e; _ -> Unknow
...
+  gasOut <- use $ hasLens . state . gas
+  cd  <- use $ hasLens . state . calldata
+  case (res, t ^. call) of
+    (f@Reversion, _) -> do
+      hasLens .= og
+      hasLens . state . calldata .= cd
+      hasLens . result ?= f
+    (VMFailure x, _) -> h x
+    (VMSuccess bc, SolCreate _) ->
+      (hasLens %=) . execState $ do
```
Comment (ggrieco-tob) [inline lib/Echidna/Exec.hs:83]: Should we remove the use of `%=` here?

File: lib/Echidna/Solidity.hs [hunk 2]
```diff
@@ -258,8 +261,10 @@ loadWithCryticCompile fp name = contracts fp >>= loadSpecified name
 -- for running a 'Campaign' against the tests found.
 prepareForTest :: (MonadReader x m, Has SolConf x)
                => (VM, NE.NonEmpty SolSignature, [Text], M.HashMap BS.ByteString (NE.NonEmpty SolSignature)) -> m (VM, World, [SolTest])
-prepareForTest (v, a, ts, m) = view hasLens <&> \SolConf { _sender = s, _checkAsserts = ch } ->
-  (v, World s m, fmap Left (zip ts $ repeat r) ++ if ch then Right <$> drop 1 a' else []) where
+prepareForTest (v, a, ts, m) = do
+  SolConf{_sender = s, _checkAsserts = ch} <- view hasLens
```
Comment (ggrieco-tob) [inline lib/Echidna/Solidity.hs:265]: spaces here before `_sender` and after `ch`

File: lib/Echidna/Exec.hs [hunk 3]
```diff
@@ -66,23 +66,26 @@ vmExcept e = throwM $ case VMFailure e of {Illegal -> IllegalExec e; _ -> Unknow
...
-                                                          hasLens . result ?= f
-                        (VMFailure x, _)            -> h x
-                        (VMSuccess bc, SolCreate _) -> (hasLens %=) . execState $ do
-                          env . contracts . at (t ^. dst) . _Just . contractcode .= InitCode ""
-                          replaceCodeOfSelf (RuntimeCode bc)
-                          loadContract (t ^. dst)
-                        _                        -> pure ()
-                      return (res, fromIntegral (gasIn - gasOut))
+execTxWith h m t = do
+  (og :: VM) <- use hasLens
```
Comment (ggrieco-tob) [inline lib/Echidna/Exec.hs:70]: I always found confusing the `og` name, where it was easier to use `vm` or something like that.

**Comment by ggrieco-tob** [comment]
This PR looks good!, my only request is include more comments when possible, in particular in the `do` statements that you modified (since this way to code makes easier to add them).

File: lib/Echidna/Exec.hs (hunk 3 above)
Comment (arcz) [reply lib/Echidna/Exec.hs:70]: good point, I will change this to `vm`

File: lib/Echidna/Exec.hs (hunk 1 above)
Comment (arcz) [reply lib/Echidna/Exec.hs:83]: I think we can leave it for now as is. I will be looking into more simplifications after we merge this PR.

Comment (ggrieco-tob) [reply lib/Echidna/Exec.hs:83]: Fair enough.

**Pull request merged**